python/ml4ir/base/data/tfrecord_writer.py

```diff
@@ -104,100 +118,94 @@ def write_from_df(
-        help="Path to feature config JSON file or feature config JSON string",
-    )
-    parser.add_argument(
-        "--convert_single_files",
-        type=bool,
-        default=False,
-        help="Whether to convert each CSV file individually"
-        "All occurences of a query key should be within a single file",
-    )
-    args = parser.parse_args(argv)
-
-    # Get all CSV files to be converted
+    # Get all CSV files to be converted, depending on user's arguments
     if args.csv_dir:
         csv_files: List[str] = glob.glob(os.path.join(args.csv_dir, "*.csv"))
     else:
-        csv_files: List[str] = [args.csv_file]
-
-    feature_config: FeatureConfig = parse_config(args.feature_config)
-
-    # Setup logging
-    logger: Logger = setup_logging()
+        csv_files: List[str] = args.csv_files
+    # Load feat config
+    feature_config: FeatureConfig = parse_config(MODES[args.tfmode], args.feature_config, logger)
 
     # Convert to TFRecord SequenceExample protobufs and save
-    file_count = 0
-    if args.convert_single_files:
+    if args.keep_single_files:
         # Convert each CSV file individually - better performance
         for csv_file in csv_files:
-            if args.tfrecord_dir:
-                tfrecord_file: str = os.path.join(
-                    args.tfrecord_dir, "file_{}.tfrecord".format(file_count)
-                )
-            else:
-                tfrecord_file: str = args.tfrecord_file
-
+            tfrecord_file: str = os.path.basename(csv_file).replace(".csv", '')
+            tfrecord_file: str = os.path.join(args.out_dir, "{}.tfrecord".format(tfrecord_file))
             write_from_files(
                 csv_files=[csv_file],
                 tfrecord_file=tfrecord_file,
                 feature_config=feature_config,
                 logger=logger,
+                tfrecord_type=MODES[args.tfmode]
             )
 
-            file_count += 1
     else:
         # Convert all CSV files at once - expensive groupby operation
-        if args.tfrecord_dir:
-            tfrecord_file: str = os.path.join(
-                args.tfrecord_dir, "file_{}.tfrecord".format(file_count)
-            )
-        else:
-            tfrecord_file: str = args.tfrecord_file
-
+        tfrecord_file: str = os.path.join(args.out_dir, "combined.tfrecord")
         write_from_files(
             csv_files=csv_files,
             tfrecord_file=tfrecord_file,
             feature_config=feature_config,
             logger=logger,
+            tfrecord_type=MODES[args.tfmode]
         )
 
 
+def define_arguments():
+    first_doc_line = __doc__.strip().split("\n")[0]
+    parser = ArgumentParser(description=first_doc_line)
+    parser.add_argument(
+        'tfmode',
+        choices=MODES,  # Choices with a dict, shows the keys
+        help="select between `sequence` and `example` to write tf.Example or tf.SequenceExample")
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
```

Comment:
This is an important change. I am adding support for converting several files at once that are explicitly listed or all csv files of a dir. But either the one, or the other, both does not work and the code returns an error..

Reply:
The support was already there, right?

Reply:
Okay. I see what you are doing. Did not know you could define mutually exclusive arguments. Nice! I like it.

Reply:
Yes the support was there. The change just make it explicit in the arguments (like you can have either one or the other). If you try to pass both, it fails giving you a clear failure message. 

The other change is in `--csv-files` where you can pass several files, space separated. This simplifies the code in line 140.